demo/src/main/java/com/novoda/demo/DialogCreator.java

```diff
@@ -0,0 +1,125 @@
+package com.novoda.demo;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.widget.ArrayAdapter;
+
+import com.novoda.noplayer.NoPlayer;
+import com.novoda.noplayer.model.AudioTracks;
+import com.novoda.noplayer.model.PlayerAudioTrack;
+import com.novoda.noplayer.model.PlayerSubtitleTrack;
+import com.novoda.noplayer.model.PlayerVideoTrack;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+class DialogCreator {
+
+    private static final String VIDEO_TRACK_MESSAGE_FORMAT = "ID: %s Quality: %s";
+    private static final String AUDIO_TRACK_MESSAGE_FORMAT = "ID: %s Type: %s";
+
+    private final Context context;
+    private final NoPlayer noPlayer;
+
+    DialogCreator(Context context, NoPlayer noPlayer) {
+        this.context = context;
+        this.noPlayer = noPlayer;
+    }
+
+    void showVideoSelectionDialog() {
+        final List<PlayerVideoTrack> videoTracks = noPlayer.getVideoTracks();
+
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(context, R.layout.list_item);
+        adapter.addAll(mapVideoTrackToLabel(videoTracks));
+        AlertDialog videoTrackSelectionDialog = new AlertDialog.Builder(context)
+                .setTitle("Select Video track")
+                .setAdapter(adapter, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int position) {
+                        if (position == 0) {
+                            noPlayer.clearVideoTrackSelection();
+                        } else {
+                            PlayerVideoTrack videoTrack = videoTracks.get(position - 1);
+                            noPlayer.selectVideoTrack(videoTrack);
+                        }
+                    }
+                }).create();
+        videoTrackSelectionDialog.show();
```

Comment:
Why not chaining to `create()`? `videoTrackSelectionDialog` is not used elsewhere.

Reply:
3fd8a49c90d98764d1def664a4d6d51739a6e9ef